chipsec/modules/common/cpu/ia_untrusted.py

```diff
@@ -28,19 +28,42 @@ def __init__(self):
         BaseModule.__init__(self)
 
     def is_supported(self):
-        if self.cs.is_register_defined('MSR_BIOS_DONE') and self.cs.register_has_field('MSR_BIOS_DONE', 'IA_UNTRUSTED'):
+        if self.cs.is_register_defined('MSR_BIOS_DONE') and \
+           self.cs.register_has_field('MSR_BIOS_DONE', 'IA_UNTRUSTED'):
             return True
         self.res = ModuleResult.NOTAPPLICABLE
         return False
 
+    def check_untrusted(self):
+        self.logger.log('[*] Check that untrusted mode has been set.')
+        res = ModuleResult.PASSED
+        if self.cs.register_has_field('MSR_BIOS_DONE', 'SoC_BIOS_DONE'):
+            soc = self.cs.read_register_field('MSR_BIOS_DONE', 'SoC_BIOS_DONE')
+            if soc == 0:
+                res = ModuleResult.FAILED
+                self.logger.log_bad('SoC_BIOS_DONE not set.')
+            else:
+                self.logger.log_good('SoC_BIOS_DONE set.')
+
+        self.logger.log("")
+        for tid in range(self.cs.msr.get_cpu_thread_count()):

```

Comment:
We really should have a helper to verify a register is programmed the same way on every instance.

Reply:
Agree